ztoc/compression/tar_zinfo.go

```diff
@@ -0,0 +1,198 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package compression
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"os"
+)
+
+const (
+	// byte length of `TarZinfo`: version(4) + spanSize(8) + tar size(8)
+	tarZinfoLength = 4 + 8 + 8
+	// `TarZinfo` version. consistent with `GzipZinfo` version
+	zinfoVersion = 2
+)
+
+// TarZinfo implements the `Zinfo` interface for uncompressed tar files.
+// It only needs a span size and tar file size, since a tar file is already
+// uncompressed.
+// For tar file, `compressed`-related concepts (e.g., `CompressedArchiveSize`)
+// are only to santisfy the `Zinfo` interface and equal to their `uncompressed`-equivalent.
+type TarZinfo struct {
+	version  int32
+	spanSize int64
+	size     int64
+}
+
+// newTarZinfo creates a new instance of `TarZinfo` from serialized bytes.
+func newTarZinfo(zinfoBytes []byte) (*TarZinfo, error) {
+	if len(zinfoBytes) != tarZinfoLength {
```

Comment:
Should we model this as a flatbuf? We didn't for gzip, but that was probably a mistake.

Reply:
You mean model a corresponding table in `ztoc.fbs` (like `table TarZinfo`)?

I'm not sure if `fbs` itself supports interface abstraction (i.e., a single `Zinfo` points to either a `GzipZinfo` or a `TarZinfo` table, or `ZstdZinfo` in the future).

Also this probably requires more (possibly breaking) changes on the fbs definition and `flatbuf <-> Ztoc struct` conversion. And we'll face one zinfo `GzipZinfo` being `[]byte` and another one `TarZinfo` being a strut or a fbs table.

WDYT?

Reply:
No, I mean modeling it in compression. We could still serialize it into the ztoc as `[]byte`, but serializing/deserializing into that `[]byte` would be easier to evolve over time.